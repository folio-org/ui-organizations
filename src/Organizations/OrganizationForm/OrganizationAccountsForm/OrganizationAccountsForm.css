.panels {
  background: var(--panel-bg-color);
  padding: 1.2rem 1.2rem 0;
  border-radius: var(--panel-radius);
  border-bottom: var(--panel-border-bottom);
}

.subHeadings {
  color: var(--subHeading-color);
  font-size: var(--subHeading-font-size);
  font-weight: var(--subHeading-font-weight);
  margin: .3rem auto 1em;
}
