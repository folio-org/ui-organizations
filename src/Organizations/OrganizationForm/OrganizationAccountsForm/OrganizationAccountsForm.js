import React, { Component } from 'react';
import { FormattedMessage } from 'react-intl';
import {
  Field,
  FieldArray,
} from 'redux-form';

import {
  Button,
  Col,
  IconButton,
  Row,
  Select,
  TextArea,
  TextField,
} from '@folio/stripes/components';
import {
  FieldSelect,
  PAYMENT_METHOD_OPTIONS,
} from '@folio/stripes-acq-components';

import { ORGANIZATION_STATUS } from '../../../common/constants';
import { Required } from '../../../Utils/Validate';
import css from './OrganizationAccountsForm.css';

class OrganizationAccountsForm extends Component {
  renderForm = ({ fields }) => {
    return (
      <Row>
        <Col xs={12}>
          {fields.length === 0 &&
            <div>
              <em>
                {<FormattedMessage id="ui-organizations.accounts.pleaseAddAccount" />}
              </em>
            </div>
          }
          {fields.map(this.renderSubForm)}
        </Col>
        <Col
          xs={12}
          style={{ paddingTop: '10px' }}
        >
          <Button
            data-test-add-account-button
            onClick={() => fields.push({})}
          >
            {<FormattedMessage id="ui-organizations.accounts.add" />}
          </Button>
        </Col>
      </Row>
    );
  };

  renderSubForm = (elem, index, fields) => {
    return (
      <div
        className={css.panels}
        key={index}
      >
        <Row key={index}>
          <Col
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              label={<FormattedMessage id="ui-organizations.accounts.name" />}
              name={`${elem}.name`}
              required
              validate={[Required]}
            />
          </Col>
          <Col
            xs={6}
            md={3}
          >
            <Field
              label={<FormattedMessage id="ui-organizations.accounts.accountNumber" />}
              name={`${elem}.accountNo`}
              validate={[Required]}
              required
              component={TextField}
              fullWidth
            />
          </Col>
          <Col
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              label={<FormattedMessage id="ui-organizations.accounts.description" />}
              name={`${elem}.description`}
            />
          </Col>
          <Col
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              label={<FormattedMessage id="ui-organizations.accounts.payable" />}
              name={`${elem}.appSystemNo`}
            />
          </Col>
          <Col
            xs={6}
            md={3}
          >
            <FieldSelect
              dataOptions={PAYMENT_METHOD_OPTIONS}
              label={<FormattedMessage id="ui-organizations.accounts.paymentMethod" />}
              name={`${elem}.paymentMethod`}
              required
            />
          </Col>
          <Col
            xs={6}
            md={3}
          >
            <Field
              component={Select}
              fullWidth
              label={<FormattedMessage id="ui-organizations.accounts.account.accountStatus" />}
              name={`${elem}.accountStatus`}
              placeholder=" "
              required
              validate={[Required]}
            >
              {Object.keys(ORGANIZATION_STATUS).map((key) => (
                <FormattedMessage
                  id={`ui-organizations.organizationStatus.${key}`}
                  key={key}
                >
                  {(message) => <option value={ORGANIZATION_STATUS[key]}>{message}</option>}
                </FormattedMessage>
              ))}
            </Field>
          </Col>
          <Col
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              id={`${elem}.contactInfo`}
              label={<FormattedMessage id="ui-organizations.accounts.account.contactInfo" />}
              name={`${elem}.contactInfo`}
            />
          </Col>
          <Col
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              label={<FormattedMessage id="ui-organizations.accounts.libraryCode" />}
              name={`${elem}.libraryCode`}
              required
              validate={[Required]}
            />
          </Col>
          <Col
            xs={6}
            md={3}
          >
            <Field
              component={TextField}
              fullWidth
              label={<FormattedMessage id="ui-organizations.accounts.libraryEDICode" />}
              name={`${elem}.libraryEdiCode`}
              required
              validate={[Required]}
            />
          </Col>
          <Col
            xs={6}
            md={3}
          >
            <Field
              component={TextArea}
              fullWidth
              id={`${elem}.notes`}
              label={<FormattedMessage id="ui-organizations.accounts.notes" />}
              name={`${elem}.notes`}
            />
          </Col>
          <Col
            xs={12}
            style={{ textAlign: 'right' }}
          >
            <IconButton
              data-test-remove-account-button
              icon="trash"
              onClick={() => fields.remove(index)}
            >
              {<FormattedMessage id="ui-organizations.accounts.remove" />}
            </IconButton>
          </Col>
        </Row>
      </div>
    );
  };

  render() {
    return (
      <FieldArray
        name="accounts"
        component={this.renderForm}
      />
    );
  }
}

export default OrganizationAccountsForm;
