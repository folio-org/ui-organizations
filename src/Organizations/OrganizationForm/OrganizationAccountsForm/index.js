export { default as OrganizationAccountsForm } from './OrganizationAccountsForm';
