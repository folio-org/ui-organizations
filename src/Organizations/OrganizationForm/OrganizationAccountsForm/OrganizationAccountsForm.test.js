import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import stripesFinalForm from '@folio/stripes/final-form';

import OrganizationAccountsForm from './OrganizationAccountsForm';

const TestForm = stripesFinalForm({})(
  () => {
    return (
      <form>
        <OrganizationAccountsForm />
      </form>
    );
  },
);

const account = {
  accountNo: '1234',
  accountStatus: 'Active',
  acqUnitIds: [],
  appSystemNo: '',
  contactInfo: 'service@amazon.com',
  description: 'Monographic ordering unit account',
  libraryCode: 'COB',
  libraryEdiCode: '765987610',
  name: 'Monographic ordering unit account',
  notes: '',
  paymentMethod: 'Credit Card',
};
const renderForm = ({ initialValues = {} } = {}) => render(
  <TestForm
    onSubmit={jest.fn()}
    initialValues={initialValues}
  />,
  { wrapper: MemoryRouter },
);

describe('OrganizationAccountsForm', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render correct structure with defined accounts', async () => {
    const { asFragment } = renderForm({ initialValues: { accounts: [account] } });

    await screen.findByText('ui-organizations.accounts.name');

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correct structure without accounts', async () => {
    const { asFragment } = renderForm();

    expect(asFragment()).toMatchSnapshot();
  });
});
