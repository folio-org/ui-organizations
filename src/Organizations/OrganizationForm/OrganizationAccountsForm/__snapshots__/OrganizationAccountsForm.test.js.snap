// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationAccountsForm should render correct structure with defined accounts 1`] = `
<DocumentFragment>
  <form>
    <div
      class="row"
    >
      <div
        class="col-xs-12"
      >
        <div
          class="panels"
        >
          <div
            class="row"
          >
            <div
              class="col-xs-6 col-md-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="text-input-1"
                >
                  ui-organizations.accounts.name
                  <span
                    aria-hidden="true"
                    class="asterisk"
                  >
                    *
                  </span>
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    aria-required="true"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="text-input-1"
                    name="accounts[0].name"
                    required=""
                    spellcheck="false"
                    type="text"
                    value="Monographic ordering unit account"
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-6 col-md-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="text-input-2"
                >
                  ui-organizations.accounts.accountNumber
                  <span
                    aria-hidden="true"
                    class="asterisk"
                  >
                    *
                  </span>
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    aria-required="true"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="text-input-2"
                    name="accounts[0].accountNo"
                    required=""
                    spellcheck="false"
                    type="text"
                    value="1234"
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-6 col-md-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="text-input-3"
                >
                  ui-organizations.accounts.description
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="text-input-3"
                    name="accounts[0].description"
                    spellcheck="false"
                    type="text"
                    value="Monographic ordering unit account"
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-6 col-md-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="text-input-4"
                >
                  ui-organizations.accounts.payable
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="text-input-4"
                    name="accounts[0].appSystemNo"
                    spellcheck="false"
                    type="text"
                    value=""
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-6 col-md-3"
            >
              <div
                class="select inputGroup fullWidth"
              >
                <label
                  class="label"
                  for="select-5"
                  id="select-5-label"
                >
                  ui-organizations.accounts.paymentMethod
                </label>
                <div
                  class="selectWrap"
                >
                  <select
                    aria-invalid="false"
                    aria-labelledby="select-5-label"
                    aria-required="false"
                    class="selectControl formControl"
                    id="select-5"
                    name="accounts[0].paymentMethod"
                  >
                    <option
                      value=""
                    >
                      stripes-acq-components.label.emptyValue
                    </option>
                    <option
                      value="Cash"
                    >
                      stripes-acq-components.paymentMethod.cash
                    </option>
                    <option
                      value="Credit Card"
                    >
                      stripes-acq-components.paymentMethod.card
                    </option>
                    <option
                      value="EFT"
                    >
                      stripes-acq-components.paymentMethod.eft
                    </option>
                    <option
                      value="Deposit Account"
                    >
                      stripes-acq-components.paymentMethod.depAccount
                    </option>
                    <option
                      value="Physical Check"
                    >
                      stripes-acq-components.paymentMethod.physicalCheck
                    </option>
                    <option
                      value="Bank Draft"
                    >
                      stripes-acq-components.paymentMethod.bankDraft
                    </option>
                    <option
                      value="Internal Transfer"
                    >
                      stripes-acq-components.paymentMethod.internalTransfer
                    </option>
                    <option
                      value="Other"
                    >
                      stripes-acq-components.paymentMethod.other
                    </option>
                  </select>
                  <span>
                    Icon
                  </span>
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-6 col-md-3"
            >
              <div
                class="select inputGroup fullWidth"
              >
                <label
                  class="label"
                  for="select-6"
                  id="select-6-label"
                >
                  ui-organizations.accounts.account.accountStatus
                  <span
                    aria-hidden="true"
                    class="asterisk"
                  >
                    *
                  </span>
                </label>
                <div
                  class="selectWrap"
                >
                  <select
                    aria-invalid="false"
                    aria-labelledby="select-6-label"
                    aria-required="true"
                    class="selectControl formControl"
                    id="select-6"
                    name="accounts[0].accountStatus"
                    required=""
                  >
                    <option
                      disabled=""
                      value=""
                    >
                       
                    </option>
                    <option
                      value="Active"
                    >
                      ui-organizations.organizationStatus.active
                    </option>
                    <option
                      value="Inactive"
                    >
                      ui-organizations.organizationStatus.inactive
                    </option>
                    <option
                      value="Pending"
                    >
                      ui-organizations.organizationStatus.pending
                    </option>
                  </select>
                  <span>
                    Icon
                  </span>
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-6 col-md-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="accounts[0].contactInfo"
                >
                  ui-organizations.accounts.account.contactInfo
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="accounts[0].contactInfo"
                    name="accounts[0].contactInfo"
                    spellcheck="false"
                    type="text"
                    value="service@amazon.com"
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-6 col-md-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="text-input-7"
                >
                  ui-organizations.accounts.libraryCode
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="text-input-7"
                    name="accounts[0].libraryCode"
                    spellcheck="false"
                    type="text"
                    value="COB"
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-6 col-md-3"
            >
              <div
                class="textField"
              >
                <label
                  class="label"
                  for="text-input-8"
                >
                  ui-organizations.accounts.libraryEDICode
                </label>
                <div
                  class="formControl inputGroup"
                >
                  <input
                    aria-invalid="false"
                    autocapitalize="off"
                    autocomplete="off"
                    autocorrect="off"
                    error=""
                    id="text-input-8"
                    name="accounts[0].libraryEdiCode"
                    spellcheck="false"
                    type="text"
                    value="765987610"
                    warning=""
                  />
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-6 col-md-3"
            >
              <div
                class="textArea inputGroup fullWidth"
              >
                <label
                  class="label"
                  for="accounts[0].notes"
                >
                  ui-organizations.accounts.notes
                </label>
                <textarea
                  aria-invalid="false"
                  class="formControl"
                  id="accounts[0].notes"
                  name="accounts[0].notes"
                  type="text"
                />
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-6 col-md-3"
            >
              <div
                class="multiSelectContainer"
                id="account-acq-units-0"
              >
                <div
                  aria-live="assertive"
                  aria-relevant="additions"
                  class="sr-only"
                />
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="account-acq-units-0-label"
                  class="multiSelectContainer"
                  role="application"
                >
                  <label
                    class="label"
                    for="account-acq-units-0-input"
                    id="account-acq-units-0-label"
                  >
                    stripes-acq-components.label.acqUnits
                  </label>
                  <span
                    class="sr-only"
                    id="multi-value-status-account-acq-units-0"
                  >
                    0 items selected
                  </span>
                  <span
                    class="sr-only"
                    id="multi-describe-control-account-acq-units-0"
                  >
                    stripes-components.multiSelection.controlDescription
                  </span>
                  <div
                    class="multiSelectControlWrapper"
                  >
                    <div
                      aria-describedby="multi-describe-control-account-acq-units-0"
                      aria-labelledby="account-acq-units-0-label multi-value-status-account-acq-units-0"
                      class="multiSelectControl formControl"
                      role="search"
                      tabindex="0"
                    >
                      <div
                        class="multiSelectControlGroup"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-labelledby="account-acq-units-0-label"
                          autocomplete="off"
                          class="multiSelectFilterField"
                          id="account-acq-units-0-input"
                          type="text"
                          value=""
                        />
                        <input
                          aria-autocomplete="list"
                          aria-hidden="true"
                          autocomplete="off"
                          class="multiSelectValueInput"
                          id="multiselect-input-account-acq-units-0"
                          tabindex="-1"
                          type="text"
                          value=""
                        />
                      </div>
                      <button
                        aria-haspopup="true"
                        aria-label="open menu"
                        class="multiSelectToggleButton"
                        data-toggle="true"
                        role="button"
                        type="button"
                      >
                        <div
                          class="textFieldIcon"
                        >
                          <span>
                            Icon
                          </span>
                        </div>
                      </button>
                    </div>
                    <div
                      class="multiSelectMenu"
                      hidden=""
                      style="width: 0px;"
                    >
                      <div
                        class="multiSelectFeedback"
                        role="alert"
                      >
                        <div
                          class="multiSelectEmptyMessage"
                        >
                          stripes-components.multiSelection.defaultEmptyMessage
                        </div>
                      </div>
                      <ul
                        aria-labelledby="account-acq-units-0-label"
                        class="multiSelectOptionList"
                        id="multiselect-option-list-account-acq-units-0"
                        role="listbox"
                        style="max-height: 168px;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  role="alert"
                />
              </div>
            </div>
            <div
              class="col-xs-12"
              style="text-align: right;"
            >
              <button
                aria-label="trash"
                class="iconButton medium"
                data-test-remove-account-button="true"
                icon="trash"
                type="button"
              >
                <span
                  class="iconButtonInner mediumOffset"
                  data-test-remove-account-button="true"
                >
                  <span>
                    Icon
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-xs-12"
        style="padding-top: 10px;"
      >
        <button
          class="button default"
          data-test-add-account-button="true"
          type="button"
        >
          <span
            class="inner"
          >
            ui-organizations.accounts.add
          </span>
        </button>
      </div>
    </div>
  </form>
  <div />
</DocumentFragment>
`;

exports[`OrganizationAccountsForm should render correct structure without accounts 1`] = `
<DocumentFragment>
  <form>
    <div
      class="row"
    >
      <div
        class="col-xs-12"
      >
        <div>
          <em>
            ui-organizations.accounts.pleaseAddAccount
          </em>
        </div>
      </div>
      <div
        class="col-xs-12"
        style="padding-top: 10px;"
      >
        <button
          class="button default"
          data-test-add-account-button="true"
          type="button"
        >
          <span
            class="inner"
          >
            ui-organizations.accounts.add
          </span>
        </button>
      </div>
    </div>
  </form>
  <div />
</DocumentFragment>
`;
